src/plugins/data/public/ui/query_editor/query_editor.tsx

```diff
@@ -186,6 +186,12 @@ export const QueryEditorUI: React.FC<Props> = (props) => {
 
   const onSelectLanguage = (languageId: string) => {
     const newQuery = queryString.getInitialQueryByLanguage(languageId);
+    if (newQuery.dataset) {
```

Comment:
i might need a little bit more context here. within the query editor this newQuery will always have a dataset defined. 

the getInitialQueryByDataset seems to be doing the same thing as well.

Reply:
Can be a fast follow. We should remove langugae fromt he dataset and pass the query object to the dataset config instead

Reply:
Fixed